homeassistant/components/decora_wifi/light.py

```diff
@@ -2,90 +2,52 @@
 
 import logging
 
-# pylint: disable=import-error
-from decora_wifi import DecoraWiFiSession
-from decora_wifi.models.person import Person
-from decora_wifi.models.residence import Residence
-from decora_wifi.models.residential_account import ResidentialAccount
-import voluptuous as vol
-
 from homeassistant.components.light import (
     ATTR_BRIGHTNESS,
     ATTR_TRANSITION,
-    PLATFORM_SCHEMA,
     SUPPORT_BRIGHTNESS,
     SUPPORT_TRANSITION,
     LightEntity,
 )
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, EVENT_HOMEASSISTANT_STOP
-import homeassistant.helpers.config_validation as cv
-
-_LOGGER = logging.getLogger(__name__)
 
-# Validation of the user's configuration
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Required(CONF_USERNAME): cv.string, vol.Required(CONF_PASSWORD): cv.string}
+from .common import (
+    DecoraWifiCommFailed,
+    DecoraWifiEntity,
+    DecoraWifiPlatform,
+    DecoraWifiSessionNotFound,
 )
+from .const import CONF_LIGHT, DOMAIN
 
-NOTIFICATION_ID = "leviton_notification"
-NOTIFICATION_TITLE = "myLeviton Decora Setup"
+_LOGGER = logging.getLogger(__name__)
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Decora WiFi platform."""
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Decora Wifi lights based on a config entry."""
 
-    email = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-    session = DecoraWiFiSession()
+    email = hass.data[DOMAIN][entry.entry_id]
```

Comment:
change to get the emailadres from the entry data, see comment above.

Reply:
Corrected.